core-jvm-tests/src/test/scala/org/http4s/otel4s/middleware/ClientMiddlewareTests.scala

```diff
@@ -62,17 +71,20 @@ class ClientMiddlewareTests extends CatsEffectSuite {
           assertEquals(span.getKind, JSpanKind.CLIENT)
 
           val attributes = span.getAttributes.toScala
+          assertEquals(attributes.size, 11)
           def getAttr[A: AttributeKey.KeySelect](name: String): Option[A] =
             attributes.get[A](name).map(_.value)
 
           assertEquals(getAttr[String]("http.request.method"), Some("GET"))
+          assertEquals(getAttr[Seq[String]]("http.request.header.foo"), Some(Seq("bar")))
```

Comment:
Would it make sense to explicitly assert that `baz` is not here?  (And similarly `foo` on the response?)  I guess that's implied by the attributes size assertion, but it took me a second look to follow the test.